Java/jni/native_document.cc

```diff
@@ -262,7 +262,11 @@ JNIEXPORT jint JNICALL Java_com_couchbase_cbforest_Document_insertRevisionWithHi
     {
         // Convert jhistory, a Java String[], to a C array of C4Slice:
         jsize n = env->GetArrayLength(jhistory);
+#ifdef _MSC_VER
+        C4Slice* history = new C4Slice[n];
```

Comment:
If we're going to use dynamic allocation, this should be a vector -- `std::vector<C4Slice> history(n)`.